x-pack/plugins/actions/server/lib/get_oauth_access_token.ts

```
@@ -0,0 +1,89 @@
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { Logger } from '@kbn/core/server';
+import { GoogleAuth } from 'google-auth-library';
+import { ConnectorToken, ConnectorTokenClientContract } from '../types';
+
+interface GetOAuthJwtAccessTokenOpts {
+  connectorId?: string;
+  logger: Logger;
+  credentials: string;
+  connectorTokenClient?: ConnectorTokenClientContract;
+}
+
+export const getOAuthJwtAccessToken = async ({
+  connectorId,
+  logger,
+  credentials,
+  connectorTokenClient,
+}: GetOAuthJwtAccessTokenOpts) => {
+  let accessToken: string | null | undefined;
+  let connectorToken: ConnectorToken | null = null;
+  let hasErrors: boolean = false;
+  let credentialsJSON;
+  const expiresInSec = 3600;
+
+  if (connectorId && connectorTokenClient) {
+    // Check if there is a token stored for this connector
+    const { connectorToken: token, hasErrors: errors } = await connectorTokenClient.get({
+      connectorId,
+    });
+
+    connectorToken = token;
+    hasErrors = errors;
+  }
+
+  if (connectorToken === null || Date.parse(connectorToken.expiresAt) <= Date.now()) {
+    // Save the time before requesting token so we can use it to calculate expiration
+    const requestTokenStart = Date.now();
+
+    // Validate the service account credentials JSON file input
+    try {
+      credentialsJSON = JSON.parse(credentials);
+    } catch (error) {
+      return `Failed to parse credentials JSON file: Invalid JSON format: ${error.message ?? ''}`;
+    }
+
+    // request access token with service account credentials file
+    const auth = new GoogleAuth({
+      credentials: credentialsJSON,
+      scopes: 'https://www.googleapis.com/auth/cloud-platform',
+    });
+
+    try {
+      accessToken = await auth.getAccessToken();
+    } catch (error) {
+      return `Unable to retreive access token: ${error.message}`;
```

Since this function returns the access token as a string, the following code will try to use this string as an access token. Instead, we should throw an error here

---

Required changes has been done.